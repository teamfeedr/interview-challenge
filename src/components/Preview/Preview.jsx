import React from 'react';
import PropTypes from 'prop-types';

import MenuItem from '../MenuItem/MenuItem';

const getPreviewItemElements = ({ selectedItems, handleDeselectItem }) => {
    return selectedItems.map((item) => (
        <MenuItem
            key={item.name}
            id={item.id}
            name={item.name}
            dietaries={item.dietaries}
            handleOnDeselect={() => {
                handleDeselectItem({ item: { ...item } });
            }}
        />
    ));
}

const Preview = ({ selectedItems, handleDeselectItem }) => {
    const menuItemElements = getPreviewItemElements({ selectedItems, handleDeselectItem });

    return (
        <>
            <h2>Menu preview</h2>
            <ul className="menu-preview">
                {menuItemElements}
            </ul>
        </>
    )
}

Preview.propTypes = {
    selectedItems: PropTypes.arrayOf(PropTypes.object).isRequired,
    handleDeselectItem: PropTypes.func.isRequired
}

export default Preview;