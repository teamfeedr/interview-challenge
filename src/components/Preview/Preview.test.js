import React from "react";
import { shallow } from "enzyme";

import Preview from "./Preview";

describe("the Preview component", () => {
  it("renders with selectedItems", () => {
    const handleOnDeselect = () => {};
    const selectedItems = [
      {
        id: 1,
        name: "test1",
        dietaries: ["v", "vg"],
        handleOnDeselect
      },
      {
        id: 2,
        name: "test2",
        dietaries: ["vg", "gf"],
        handleOnDeselect
      }
    ];

    const rendered = shallow(
      <Preview selectedItems={selectedItems} handleDeselectItem={() => {}} />
    );

    expect(rendered).toMatchSnapshot();
  });
});
