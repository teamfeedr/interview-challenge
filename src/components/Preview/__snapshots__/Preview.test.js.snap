// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Preview component renders with selectedItems 1`] = `
<Fragment>
  <h2>
    Menu preview
  </h2>
  <ul
    className="menu-preview"
  >
    <MenuItem
      dietaries={
        Array [
          "v",
          "vg",
        ]
      }
      handleOnDeselect={[Function]}
      id={1}
      key="test1"
      name="test1"
    />
    <MenuItem
      dietaries={
        Array [
          "vg",
          "gf",
        ]
      }
      handleOnDeselect={[Function]}
      id={2}
      key="test2"
      name="test2"
    />
  </ul>
</Fragment>
`;
